scripts/plugin_tester.ts

```diff
@@ -0,0 +1,103 @@
+import { BackTester__factory } from '@typechain/index'
+import { BigNumber, providers, utils } from 'ethers'
+import { BytesLike, formatEther } from 'ethers/lib/utils'
+import { ethers } from 'hardhat'
+import backtesterArtifact from '../artifacts/contracts/backtester/BackTester.sol/BackTester.json'
+import { bn, fp } from '#/common/numbers'
+import { networkConfig } from '#/common/configuration'
+import { oracleTimeout } from './deployment/utils'
+
+export const backtestPlugin = async (deploymentTransactionData: BytesLike, stride: number, numberOfSamples: number) => {
+  const fakeCollateralAddr = ethers.Wallet.createRandom().address
+  const backTesterInterface = BackTester__factory.createInterface()
+  const provider = new providers.JsonRpcProvider(process.env.MAINNET_RPC_URL)
+  const currentBlock = await provider.getBlockNumber()
+
+  const start = currentBlock - numberOfSamples * stride
+
+  const samples: Array<{
+    block: number
+
+    low: string
+    high: string
+    refPrTok: string
+
+    status: "SOUND" | "FIFFY" | "DISABLED"
+    defaulted: boolean
+  }> = []
+
+  const state = {
+    previousStatus: 0,
+    whenDefault: 2n ** 48n - 1n,
+    defaulted: false
+  }
+  for (let i = 0; i < numberOfSamples; i++) {
+    const block = start + i * stride
+    const backtestCall = backTesterInterface.encodeFunctionData('backtestPlugin', [deploymentTransactionData, state.previousStatus, state.whenDefault])
+    const result: string = await provider.send('eth_call', [
+      {
+        to: fakeCollateralAddr,
+        data: backtestCall,
+        from: '0xaC5dfd1990ff6abA9E6F0eDA4BbeCc3be676F09A',
+      },
+      '0x' + block.toString(16),
+      {
+        [fakeCollateralAddr]: {
+          code: backtesterArtifact.deployedBytecode,
+        },
+      },
+    ])
+    const { low, high, refPrTok, status, whenDefault } = backTesterInterface.decodeFunctionResult(
+      'backtestPlugin',
+      result
+    )[0]
+
+    state.previousStatus = status
+    state.whenDefault = (whenDefault as BigNumber).toBigInt()
+
+    const sampleStatus = status === 0 ? "SOUND" : status === 1 ? "FIFFY" : "DISABLED"
+    samples.push({
+      block,
+      low: formatEther(low),
+      high: formatEther(high),
+      refPrTok: formatEther(refPrTok),
+      status: sampleStatus,
+      defaulted: status === 2,
+    })
+
+    if (status === 2) {
+      break
+    }
+  }
+  return samples
+}
+
+export const main = async () => {
+  const PRICE_TIMEOUT = bn('604800') // 1 week
+  const REVENUE_HIDING = fp('0') // no revenue hiding by default; test individually
```

Comment:
i think this makes sense, but it would be great to be able to run the simulation of a range of revenue hiding options and see how well the plugin does (0.01%, 0.001%, 0.0001%, etc...). maybe we take this functionality up 1 level and pass the revenueHiding variable in as an environment variable so we can run the same script in parallel in different threads